flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -684,6 +689,96 @@ private void setCoLocation() {
+
+		for (Entry<Integer, JobVertex> entry : jobVertices.entrySet()) {
+			final int headOperatorId = entry.getKey();
+			final JobVertex jobVertex = entry.getValue();
+
+			final SlotSharingGroup jobVertexSlotSharingGroup = jobVertex.getSlotSharingGroup();
+			if (jobVertexSlotSharingGroup != null) {
+				slotSharingGroups.add(jobVertexSlotSharingGroup);
+			}
+
+			vertexHeadOperators.put(jobVertex.getID(), headOperatorId);
+
+			final Set<Integer> operatorIds = new HashSet<>();
+			operatorIds.add(headOperatorId);
+			operatorIds.addAll(chainedConfigs.getOrDefault(headOperatorId, new HashMap<>()).keySet());
+			vertexOperators.put(jobVertex.getID(), operatorIds);
+		}
+
+		for (SlotSharingGroup slotSharingGroup : slotSharingGroups) {
+			setManagedMemoryFractionForSlotSharingGroup(slotSharingGroup, vertexHeadOperators, vertexOperators);
+		}
+	}
+
+	private void setManagedMemoryFractionForSlotSharingGroup(
+			final SlotSharingGroup slotSharingGroup,
+			final Map<JobVertexID, Integer> vertexHeadOperators,
+			final Map<JobVertexID, Set<Integer>> vertexOperators) {
+
+		int groupOperatorCount = 0;
+		for (JobVertexID jobVertexID : slotSharingGroup.getJobVertexIds()) {
+			final Set<Integer> operatorIds = vertexOperators.get(jobVertexID);
+			groupOperatorCount += operatorIds.size();
+		}
+
+		for (JobVertexID jobVertexID : slotSharingGroup.getJobVertexIds()) {
+			for (int operatorNodeId : vertexOperators.get(jobVertexID)) {
+				final StreamConfig operatorConfig = vertexConfigs.get(operatorNodeId);
+				final ResourceSpec operatorResourceSpec = streamGraph.getStreamNode(operatorNodeId).getMinResources();
+				setManagedMemoryFractionForOperator(
+					operatorResourceSpec,
+					slotSharingGroup.getResourceSpec(),
+					groupOperatorCount,
+					operatorConfig);
+			}
+
+			// need to refresh the chained task configs
+			final int headOperatorNodeId = vertexHeadOperators.get(jobVertexID);
+			final StreamConfig vertexConfig = vertexConfigs.get(headOperatorNodeId);
+			vertexConfig.setTransitiveChainedTaskConfigs(chainedConfigs.get(headOperatorNodeId));
+		}
+	}
+
+	private void setManagedMemoryFractionForOperator(
+			final ResourceSpec operatorResourceSpec,
+			final ResourceSpec groupResourceSpec,
+			final int groupOperatorCount,
+			final StreamConfig operatorConfig) {
+
+		final double managedMemoryFractionOnHeap;
+		final double managedMemoryFractionOffHeap;
+
+		if (groupResourceSpec.equals(ResourceSpec.UNKNOWN)) {
+			checkArgument(groupOperatorCount > 0, "A slot sharing group must contain at least 1 operator");
+
+			managedMemoryFractionOnHeap = 1.0 / groupOperatorCount;
+			managedMemoryFractionOffHeap = 1.0 / groupOperatorCount;
+		} else {
+			final long groupOnHeapManagedMemoryBytes = groupResourceSpec.getOnHeapManagedMemory().getBytes();
+			final long groupOffHeapManagedMemoryBytes = groupResourceSpec.getOffHeapManagedMemory().getBytes();
+
+			managedMemoryFractionOnHeap = groupOnHeapManagedMemoryBytes > 0 ?
```

Comment:
let's change the formatting to
```
condition
	? A
	: B;
```
since it is sooooo much easier to read.

Reply:
done.